### PR TITLE
replace `cal_stress` in GPU relax calculation，use NUM_STREAM in INPUT file and fix memory leak in force calculation

### LINKED ISSUE #4199: Use `cal_stress` in GPU Relax calculation
### Describe the bug

in GPU calculation,should use `cal_stress` and `cal_force` in relax calculation,it's not reasonable.

### Expected behavior

_No response_

### To Reproduce

_No response_

### Environment

_No response_

### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Verify the issue is not a duplicate.
- [X] Describe the bug.
- [ ] Steps to reproduce.
- [ ] Expected behavior.
- [X] Error message.
- [ ] Environment details.
- [ ] Additional context.
- [ ] Assign a priority level (low, medium, high, urgent).
- [ ] Assign the issue to a team member.
- [ ] Label the issue with relevant tags.
- [ ] Identify possible related issues.
- [ ] Create a unit test or automated test to reproduce the bug (if applicable).
- [X] Fix the bug.
- [ ] Test the fix.
- [ ] Update documentation (if necessary).
- [ ] Close the issue and inform the reporter (if applicable).

### PR DESCRIPTION
**Pull request by A-006**
When the relaxation is performed, it is mandotory to use the `cal_stresss` parameter in the current version of ABACUS. However, it may not be necessary to compute the stress. The calculation using the pipeline needs to be controlled via the `NUM_STREAM` parameter, so add this parameter.
When calculating `force`, a memory leak occurred. This issue was resolved, and an instance of previously unreleased memory was also identified and addressed.
### Linked Issue
Fix #4199

### CODE REVIEW

File: docs/advanced/input_files/input-main.md [hunk 1]
```diff
@@ -913,7 +913,6 @@ These variables are used to control the numerical atomic orbitals related parame
 - **Description**: choose the number of streams in GPU when we compute the `LCAO`. According to different devices , we may have different effects.For most devices,the stream is
 enough when the number is bigger then 2.
 - **Default** : "4" 
-[back to top](#full-list-of-input-keywords)
```
Comment (mohanchen) [inline docs/advanced/input_files/input-main.md:918]: why this line was deleted?

File: source/module_hamilt_lcao/module_gint/gint.cpp [hunk 2]
```diff
@@ -132,44 +131,56 @@ void Gint::cal_gint(Gint_inout* inout)
...
-                                                     force,
-                                                     stress,
-                                                     this->nplane,
-                                                     dr,
-                                                     rcut,
-                                                     *this->gridt,
-                                                     ucell);
-                    for (int iat = 0; iat < nat; iat++)
+                    if (isforce || isstress){
+                        double *force = new double[ucell.nat * 3];
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint.cpp]: use stl vector instead, don't use new and delete here

File: source/module_hamilt_lcao/module_gint/gint_force.h [hunk 3]
```diff
@@ -92,6 +92,8 @@ void gint_gamma_force_gpu(hamilt::HContainer<double>* dm,
                           const int nczp,
                           double dr,
                           double* rcut,
+                          int isforce,
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force.h]: const int isforce?

File: source/module_hamilt_lcao/module_gint/gint_force.h [hunk 4]
```diff
@@ -92,6 +92,8 @@ void gint_gamma_force_gpu(hamilt::HContainer<double>* dm,
...
+                          int isstress,
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force.h]: const int isstress?

File: source/module_hamilt_lcao/module_gint/gint_force_gpu.cu [hunk 5]
```diff
@@ -63,6 +63,8 @@ void gint_gamma_force_gpu(hamilt::HContainer<double>* dm,
                           const int nczp,
                           double dr,
                           double* rcut,
+                          int isforce,
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force_gpu.cu]: add const

File: source/module_hamilt_lcao/module_gint/gint_force_gpu.cu [hunk 6]
```diff
@@ -63,6 +63,8 @@ void gint_gamma_force_gpu(hamilt::HContainer<double>* dm,
...
+                          int isstress,
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force_gpu.cu]: add const

File: source/module_hamilt_lcao/module_gint/grid_technique.cpp [hunk 7]
```diff
@@ -638,12 +639,14 @@ void Grid_Technique::cal_trace_lo(const UnitCell& ucell)
...
 
-void Grid_Technique::init_gpu_gint_variables(const UnitCell& ucell,const LCAO_Orbitals &orb)
+void Grid_Technique::init_gpu_gint_variables(const UnitCell& ucell,const LCAO_Orbitals &orb,const int num_stream)
 {
     if (is_malloced)
     {
         free_gpu_gint_variables(this->nat);
     }
+    nstreams = num_stream;
+    streams=new cudaStream_t[nstreams];
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/grid_technique.cpp:649]: can you avoid using new？

File: source/module_hamilt_lcao/module_gint/gint_force_gpu.cu [hunk 8]
```diff
@@ -227,17 +226,21 @@ void gint_gamma_force_gpu(hamilt::HContainer<double>* dm,
...
+            delete[] f_s_iat.iat_host;
         }
     }
-    // cudaFree(f_s_iat.stress_device);
-    // cudaFree(f_s_iat.force_device);
-    // cudaFree(f_s_iat.iat_device);
-    delete[] f_s_iat.stress_host;
-    delete[] f_s_iat.force_host;
-    delete[] f_s_iat.iat_host;
+    delete[] denstiy_mat.density_mat_h;
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force_gpu.cu:242]: can you avoid using delete?

File: source/module_hamilt_lcao/module_gint/gint_force_gpu.cu [hunk 9]
```diff
@@ -227,17 +226,21 @@ void gint_gamma_force_gpu(hamilt::HContainer<double>* dm,
...
             /* stress compute in CPU*/
-            cal_stress_add(f_s_iat, stress, cuda_block);
+            if (isstress){
+                cal_stress_add(f_s_iat, stress, cuda_block);
+            }
+            if (isforce){
+                cal_force_add(f_s_iat, force, atom_num_grid);
+            }
             iter_num++;
+            delete[] f_s_iat.stress_host;
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/gint_force_gpu.cu:237]: can you avoid using delete?

File: docs/advanced/input_files/input-main.md (hunk 1 above)
Comment (A-006) [reply docs/advanced/input_files/input-main.md:918]: Originally thought this was unnecessary, but later realized this is a format for markdown documents, so I added it back

File: source/module_hamilt_lcao/module_gint/grid_technique.cpp (hunk 7 above)
Comment (A-006) [reply source/module_hamilt_lcao/module_gint/grid_technique.cpp:649]: This is the pointer to the new pipeline, the function uses the double * type, and use new to control the number of pipelines from the input file

File: source/module_hamilt_lcao/module_gint/grid_technique.h [hunk 10]
```diff
@@ -162,8 +163,8 @@ class Grid_Technique : public Grid_MeshBall
     int atom_pair_mesh;
     int atom_pair_nbz;
 
-    const int nstreams = 4;
-    cudaStream_t streams[4];
+    int nstreams ;
```
Comment (mohanchen) [inline source/module_hamilt_lcao/module_gint/grid_technique.h]: initialize the variable

**Review by mohanchen — APPROVED** [review]

**Pull request merged**